File: java/com/joshmanisdabomb/lcc/block/HorizontalBlock.java
package com.joshmanisdabomb.lcc.block;

import net.minecraft.block.Block;
import net.minecraft.block.BlockState;
import net.minecraft.item.BlockItemUseContext;
import net.minecraft.state.StateContainer;
import net.minecraft.state.properties.BlockStateProperties;
import net.minecraft.util.Direction;

public class HorizontalBlock extends net.minecraft.block.HorizontalBlock implements LCCBlockHelper {

    public HorizontalBlock(Properties p) {
        super(p);
        this.setDefaultState(this.stateContainer.getBaseState().with(BlockStateProperties.HORIZONTAL_FACING, Direction.NORTH));
    }

    @Override
    protected void fillStateContainer(StateContainer.Builder<Block, BlockState> builder) {
        builder.add(BlockStateProperties.HORIZONTAL_FACING);
    }

    @Override
    public BlockState getStateForPlacement(BlockItemUseContext context) {
        return traitHorizontalPlacement(context);
    }

}

File: kotlin/com/joshmanisdabomb/lcc/block/HorizontalBlock.kt
package com.joshmanisdabomb.lcc.block

import com.joshmanisdabomb.lcc.directory.LCCBlocks.traitHorizontalPlacement
import net.minecraft.block.Block
import net.minecraft.block.BlockState
import net.minecraft.block.HorizontalFacingBlock
import net.minecraft.item.ItemPlacementContext
import net.minecraft.state.StateManager

class HorizontalBlock(settings: Settings) : HorizontalFacingBlock(settings) {

    override fun appendProperties(builder: StateManager.Builder<Block, BlockState>) = builder.add(FACING).let { Unit }

    override fun getPlacementState(context: ItemPlacementContext) = traitHorizontalPlacement(context)

}